pkg/graphql/schema.go

```diff
@@ -2,24 +2,58 @@ package graphql
 
 import (
 	"io"
+	"io/ioutil"
+
+	"github.com/jensneuse/abstractlogger"
+
+	"github.com/jensneuse/graphql-go-tools/pkg/ast"
+	"github.com/jensneuse/graphql-go-tools/pkg/astparser"
 )
 
 type Schema struct {
-	Content []byte
+	logger   abstractlogger.Logger
+	document ast.Document
 }
 
 func NewSchemaFromReader(reader io.Reader) (*Schema, error) {
-	return &Schema{Content: nil}, nil
+	logger := abstractlogger.NoopLogger
```

Comment:
Is defining log here the correct thing? I mean can't we give as a parameter and give gateway's logger?

Reply:
IMHO the library should have a really slick API for usage. Because of that I created the `SetLogger()` method. Dependency Injection doesn't need to happen in the constructor.

Library user can then decide if they want to set a logger or not.

Reply:
Is it common that you have to set a logger in a second function call?

Reply:
The point is while schema is created, some logs can be printed by the library. Right now, for those logs, how does user set a custom logger?

Reply:
there are no logs printed by the library on schema creation, so it's fine not to have a logger defined at his point.

I would argue the other way: Is it common when using a third-party library to be forced to provide a library when using it? I never encountered this until now- so this behavior may feel unfamiliar for the library user

Reply:
It is not forced. However, letting user to be able to set logger is common. So, I propose defining a global logger in library and if user wants to set it, he should be able to do this. For example, I need to set logger while using the library in gateway.

Reply:
After rechecking the code of the library, I have decided to remove the logger at this point because it does not add any value (it is unused).